snapcraft/internal/elf.py

```diff
@@ -82,21 +125,48 @@ def patch(self, *, elf_file: ElfFile) -> None:
         :raises snapcraft.internal.errors.PatcherError:
             raised when the elf_file cannot be patched.
         """
-        # When setting rpath for libraries is implemented, we will do more
-        # here.
-        if not elf_file.is_executable:
-            return
+        if elf_file.is_executable:
+            self._patch_interpreter(elf_file)
+        if elf_file.dependencies:
```

Comment:
The docstring of the function should be updated. It currently just mentions what will happen when it's executable, not when it has dependencies.